go/table.go

```diff
@@ -292,3 +292,283 @@ func (t *Table) GetVOffsetTSlot(slot VOffsetT, d VOffsetT) VOffsetT {
 	}
 	return VOffsetT(off)
 }
+
+// MutateBool updates a bool at the given offset.
+func (t *Table) MutateBool(off UOffsetT, n bool) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeBool) {
+		WriteBool(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateByte updates a Byte at the given offset.
+func (t *Table) MutateByte(off UOffsetT, n byte) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeByte) {
+		WriteByte(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateUint8 updates a Uint8 at the given offset.
+func (t *Table) MutateUint8(off UOffsetT, n uint8) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeUint8) {
+		WriteUint8(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateUint16 updates a Uint16 at the given offset.
+func (t *Table) MutateUint16(off UOffsetT, n uint16) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeUint16) {
+		WriteUint16(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateUint32 updates a Uint32 at the given offset.
+func (t *Table) MutateUint32(off UOffsetT, n uint32) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeUint32) {
+		WriteUint32(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateUint64 updates a Uint64 at the given offset.
+func (t *Table) MutateUint64(off UOffsetT, n uint64) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeUint64) {
+		WriteUint64(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateInt8 updates a Int8 at the given offset.
+func (t *Table) MutateInt8(off UOffsetT, n int8) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeInt8) {
+		WriteInt8(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateInt16 updates a Int16 at the given offset.
+func (t *Table) MutateInt16(off UOffsetT, n int16) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeInt16) {
+		WriteInt16(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateInt32 updates a Int32 at the given offset.
+func (t *Table) MutateInt32(off UOffsetT, n int32) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeInt32) {
+		WriteInt32(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateInt64 updates a Int64 at the given offset.
+func (t *Table) MutateInt64(off UOffsetT, n int64) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeInt64) {
+		WriteInt64(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateFloat32 updates a Float32 at the given offset.
+func (t *Table) MutateFloat32(off UOffsetT, n float32) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeFloat32) {
+		WriteFloat32(t.Bytes[off:], n)
+		return true
+	}
+
+	return false
+}
+
+// MutateFloat64 updates a Float64 at the given offset.
+func (t *Table) MutateFloat64(off UOffsetT, n float64) bool {
+	if off <= UOffsetT(len(t.Bytes)-SizeFloat64) {
```

Comment:
This appears to be a bounds check on the whole buffer? That seems rather inefficient, given that it does not protect against most writes in wrong locations. Also it returns false if it fails, which hides the error, it should really be an assertion failure (whatever that is in Go).